### PR TITLE
improve displaying detail information, GET, POST parameters

### PR DESCRIPTION
**Pull request by dennydanek**
fix issue when panel stuck and crashed all other tracy panels,
bar layout changed - bar shows only necessary information (time of ms / count of requests)
panel layout changed - added GET and POST parameters information when are available

### CODE REVIEW

File: src/SnapshotStack.php [hunk 1]
```diff
@@ -30,4 +30,12 @@ public function getSnapshots(): array
 		return $this->snapshots;
 	}
 
+	/**
+	 * @return int
+	 */
+	public function getNumberOfSnapshots(): int
```
Comment (f3l1x) [inline src/SnapshotStack.php:36]: Why is it needed? I prefer `count(this->snaphots)`

File: src/Tracy/Panel.php [hunk 2]
```diff
@@ -18,26 +19,28 @@ public function __construct(SnapshotStack $snapshotStack)
...
+			// phpcs:disable
+			$totalTime = $this->snapshotStack->getTotalTime();
+			$count = $this->snapshotStack->getNumberOfSnapshots();
+			require __DIR__ . '/templates/tab.phtml';
+		});
 	}
 
-	public function getPanel(): string
+	/**
+	 * @return string|null
```
Comment (f3l1x) [inline src/Tracy/Panel.php:31]: Phpdoc is not needed, is it?

File: composer.json [hunk 3]
```diff
@@ -22,6 +22,7 @@
     "php": ">=7.2",
     "guzzlehttp/guzzle": "^6.3.3|^7.0",
     "nette/di": "~3.0.0",
+    "nette/http": "^3.1",
```
Comment (f3l1x) [inline composer.json:25]: Why is nette/http needed?

**Comment by f3l1x** [comment]
Thanks. I've added some comments. Can you please include new screenshots?

File: src/SnapshotStack.php (hunk 1 above)
Comment (dennydanek) [reply src/SnapshotStack.php:36]: I'ts the same. Just because I used on two place - DRY

File: src/Tracy/Panel.php (hunk 2 above)
Comment (dennydanek) [reply src/Tracy/Panel.php:31]: Sure the PHPdoc is not needed. The return type was changed to string or null

if is null nothing will be displayed

File: composer.json (hunk 3 above)
Comment (dennydanek) [reply composer.json:25]: I used the `\Nette\Http\Url::parseQuery` method for display GET parameter as array

**Comment by dennydanek** [comment]
0 requests
image

1 request
image
image

File: composer.json (hunk 3 above)
Comment (f3l1x) [reply composer.json:25]: Can you use guzzle utils for that or php native functions? Add extra dependency just for tracy bar seems little bit heavy.

File: src/SnapshotStack.php (hunk 1 above)
Comment (f3l1x) [reply src/SnapshotStack.php:36]: Just replace it as I said, thank you. I don't wanna introduce more methods if it's just `count($var)`.

**Comment by f3l1x** [comment]
Can you include these screenshots to your changes? I would like to display them in the readme. Thanks.

File: Makefile [hunk 4]
```diff
@@ -6,13 +6,13 @@ install:
 qa: phpstan cs
 
 cs:
-	vendor/bin/codesniffer src tests
+	.\vendor\bin\phpcs --standard=ruleset.xml src tests
```
Comment (dennydanek) [inline Makefile:9]: you can change it back I rewrote only for my purpose

**Comment by dennydanek** [comment]
Is there anything what prevent to merge?

**Pull request closed without merging**